### PR TITLE
Improve Telegram X template

### PR DESCRIPTION
**Pull request by heokhe**
Some of the issues related to TGX template are fixed. The important ones are written below:
- `headerTabActive` is visible
- shadows won't replace with separators
- Message selection color is more transparent now

Also, the bot can accept hexacodes with not only 6, but 3, 4 or 8 digits.

### CODE REVIEW

File: handlers/custom-color.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module.exports = bot => {
-    bot.hears(/^#[\da-f]{6}$/i, async ctx => {
+    bot.hears(/^#(?:[\da-f]{2,3}){1,2}$/i, async ctx => {
```
Comment (vrumger) [inline handlers/custom-color.js]: This doesn't seem right
image

**Comment by vrumger** [comment]
> Also, the bot can accept hexacodes with not only 6, but 3, 4 or 8 digits.

1. The bot shouldn't support 8 digits (or just ignore the alpha)
2. It would need to convert 3 to 6

To convert 3 to 6, you can do something like this:

```javascript
const colors = ctx.match[0]
    .split(``)
    .map(color => color.repeat(2))
    .join(``);
```

**Review by vrumger — CHANGES_REQUESTED** [review]

File: handlers/custom-color.js (hunk 1 above)
Comment (heokhe) [reply handlers/custom-color.js]: Yes, that was a fault. It should be:
```js
/^#(?:[\da-f]{3, 4}){1,2}$/i
```
And if you want to ignore the alpha:
```js
/^#(?:[\da-f]{3}){1,2}$/i
```

**Review by vrumger — CHANGES_REQUESTED** [review]
I think it would be better to check the hex code length before adding it to the colors array instead of checking for each template.

**Review by vrumger — APPROVED** [review]

**Comment by vrumger** [comment]
👍

**Pull request merged**